pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageContainerImpl.java

```diff
@@ -283,7 +295,8 @@ public OpSendMsg createOpSendMsg() throws IOException {
             lowestSequenceId = -1L;
             return op;
         }
-        ByteBuf encryptedPayload = producer.encryptMessage(messageMetadata, getCompressedBatchMetadataAndPayload());
+        ByteBuf encryptedPayload = producer.encryptMessage(messageMetadata,
+                getCompressedBatchMetadataAndPayload(false));
```

Comment:
this change can be reverted once there's a `getCompressedBatchMetadataAndPayload()` method as explained in one of the previous comments.